File: worblehat-web/src/test/java/de/codecentric/psd/worblehat/web/formdata/GetMyBooksFormDataTest.java
package de.codecentric.psd.worblehat.web.formdata;

import org.junit.Before;
import org.junit.Test;

import static org.hamcrest.CoreMatchers.is;
import static org.hamcrest.CoreMatchers.nullValue;
import static org.junit.Assert.*;

public class GetMyBooksFormDataTest {

    private GetMyBooksFormData formData;

    @Before
    public void setUp() throws Exception {
        formData = new GetMyBooksFormData();
    }

    @Test
    public void getEmailAddressWithWhitespaces() {
        formData.setEmailAddress(" mit ");
        assertThat(formData.getEmailAddress(), is("mit"));
    }

    @Test
    public void getEmailAddressWithoutWhitespaces() {
        formData.setEmailAddress("ohne");
        assertThat(formData.getEmailAddress(), is("ohne"));
    }

    @Test
    public void getNullEmail() {
        formData.setEmailAddress(null);
        assertThat(formData.getEmailAddress(), is(nullValue()));
    }
}

File: worblehat-web/src/test/java/de/codecentric/psd/worblehat/web/formdata/GetMyBooksFormDataTest.kt
package de.codecentric.psd.worblehat.web.formdata

import org.hamcrest.CoreMatchers.`is`
import org.junit.Assert.assertThat
import org.junit.Before
import org.junit.Test

class GetMyBooksFormDataTest {

    private val formData = GetMyBooksFormData()

    @Test
    fun getEmailAddressWithWhitespaces() {
        formData.emailAddress = " mit "
        assertThat(formData.emailAddress, `is`("mit"))
    }

    @Test
    fun getEmailAddressWithoutWhitespaces() {
        formData.emailAddress = "ohne"
        assertThat(formData.emailAddress, `is`("ohne"))
    }

    @Test
    fun getNullEmail() {
        formData.emailAddress = ""
        assertThat(formData.emailAddress, `is`(""))
    }
}